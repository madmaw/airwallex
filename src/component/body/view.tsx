import * as React from 'react';
import { Observer } from 'rxjs';
import styles from './view.module.scss';

export type EventBodyRequestInvite = {
  type: 'body_request_invite',
};

export type Events = EventBodyRequestInvite;

export const BUTTON_TEXT_REQUEST = 'Requst an invite';

export function Body({
  observer,
}: {
  observer: Observer<Events>,
}) {
  const requestInvite = React.useCallback(() => {
    observer.next({
      type: 'body_request_invite',
    });
  }, [])

  return (
    <div className={styles.container}>
      <div className={styles.content}>
        <h1>A better way<br/>to enjoy every day.</h1>
        <p>
          Be the first to know when we launch.
        </p>
        <button onClick={requestInvite} autoFocus={true}>{BUTTON_TEXT_REQUEST}</button>
      </div>
    </div>
  );
}