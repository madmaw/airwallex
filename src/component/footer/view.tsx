import * as React from 'react';
import styles from './view.module.scss';

export function Footer() {
  return (
    <div className={styles.container}>
    Made with ❤ in Sydney<br/>
    © 2024 Brocolli & Co. All rights reserved.
    </div>
  )
}