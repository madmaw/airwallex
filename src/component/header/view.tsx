import * as React from 'react';
import styles from './view.module.scss';

export function Header({

}: {

}) {
  return (
    <div className={styles.container}>
      <h1>Broccoli & Co.</h1>
    </div>
  );
}