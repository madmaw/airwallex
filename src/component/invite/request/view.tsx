import * as React from 'react';
import { Observer } from 'rxjs';
import styles from './view.module.scss';

export const PLACEHOLDER_FULL_NAME = 'Full Name';
export const PLACEHOLDER_EMAIL = 'Email';
export const PLACEHOLDER_CONFIRM_EMAIL = 'Confirm Email';
export const BUTTON_TEXT_SEND = 'Send';

export type EventInviteRequestSubmit = {
  type: 'invite_request_submit',
};

export type EventInviteRequestInput = {
  type: 'invite_request_input',
  state: Pick<InviteRequestState, 'email' | 'name' | 'confirmEmail'>,
};

export type Events = 
  | EventInviteRequestSubmit
  | EventInviteRequestInput;

export type InviteRequestState = {
  name?: string,
  email?: string,
  confirmEmail?: string,
  errorMessage?: string,
  processing: boolean,
};

export function InviteRequest({
  state,
  observer,
}: {
  state: InviteRequestState,
  observer: Observer<Events>, 
}) {
  const {
    email,
    name,
    confirmEmail,
    errorMessage,
    processing,
  } = state;
  const setName = React.useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
    observer.next({
      type: 'invite_request_input',
      state: {
        ...state,
        name: e.target.value,
      }
    });
  }, [observer]);
  const setEmail = React.useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
    observer.next({
      type: 'invite_request_input',
      state: {
        ...state,
        email: e.target.value,
      }
    });
  }, [observer]);
  const setConfirmEmail = React.useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
    observer.next({
      type: 'invite_request_input',
      state: {
        ...state,
        confirmEmail: e.target.value,
      }
    });
  }, [observer]);
  const send = React.useCallback(() => {
    observer.next({
      type: 'invite_request_submit',
    });
  }, [observer]);
  return (
    <div className={styles.container}>
      <h2>Request an invite</h2>
      <input
        onChange={setName}
        value={name || ''}
        placeholder={PLACEHOLDER_FULL_NAME}
        disabled={processing}
        autoFocus={true}
        />
      <input
        onChange={setEmail}
        value={email || ''}
        placeholder={PLACEHOLDER_EMAIL}
        disabled={processing}
        />
      <input
        onChange={setConfirmEmail}
        value={confirmEmail || ''}
        placeholder={PLACEHOLDER_CONFIRM_EMAIL}
        disabled={processing}
        />
      <button
        className={styles.submit}
        onClick={send}
        disabled={processing}
        >{processing ? 'Sending, please wait...' : BUTTON_TEXT_SEND}</button>
      {errorMessage && (
        <div className={styles.error}>
          {errorMessage}
        </div>
      )}
    </div>
  );
}
