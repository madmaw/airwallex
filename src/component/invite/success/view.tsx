import * as React from 'react';
import { Observer } from 'rxjs';
import styles from './view.module.scss';

export type EventInviteSuccessDone = {
  type: 'invite_success_done',
};

export type Events = EventInviteSuccessDone;

export const BUTTON_TEXT_OK = 'OK';

export function InviteSuccess({
  observer,
}: {
  observer: Observer<Events>
}) {
  const onClose = React.useCallback(() => {
    observer.next({
      type: 'invite_success_done',     
    });
  }, [observer])
  return (
    <div className={styles.container}>
      <h2>All done!</h2>
      <p>You will be one of the first to experience Broccoli & Co. when we launch</p>
      <button onClick={onClose} autoFocus={true}>{BUTTON_TEXT_OK}</button>
    </div>
  ); 
}