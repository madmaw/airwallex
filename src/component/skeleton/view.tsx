import * as React from 'react';
import styles from './view.module.scss';
import { Observer } from 'rxjs';

export type EventSkeletonCloseDialog = {
  type: 'skeleton_close_dialog',
};

export type Events = EventSkeletonCloseDialog;

type SkeletonProps = {
  header: JSX.Element,
  footer: JSX.Element,
  body: JSX.Element,
  dialog: JSX.Element | undefined,
  observer: Observer<Events>,
};

export const CLOSE_BUTTON_ARIA_LABEL = 'Close';

export function DesktopSkeleton({
  header,
  footer,
  body,
  dialog,
  observer,
}: SkeletonProps) {
  const onClose = React.useCallback(() => {
    observer.next({
      type: 'skeleton_close_dialog',
    });
  }, [observer])

  const preventClose = React.useCallback((e: React.MouseEvent) => {
    e.stopPropagation();
    e.nativeEvent.stopImmediatePropagation();
  }, []);

  return (
    <div className={styles.container}>
      <div className={styles.header}>
        {header}
      </div>
      <div className={styles.body}>
        {body}
      </div>
      <div className={styles.footer}>
        {footer}
      </div>
      {dialog && (
        <div className={styles.overlay} onClick={onClose}>
          <div className={styles.dialog} onClick={preventClose}>
            {dialog}
            <button 
              className={styles.closeButton}
              onClick={onClose}
              aria-label={CLOSE_BUTTON_ARIA_LABEL}
              >x</button>
          </div>
        </div>
      )}
    </div>
  )
}