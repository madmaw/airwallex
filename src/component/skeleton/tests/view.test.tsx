import { RenderResult, render } from '@testing-library/react';
import * as React from 'react';
import { Mocked, afterEach, beforeEach, describe, expect, it, vi } from "vitest";
import { CLOSE_BUTTON_ARIA_LABEL, DesktopSkeleton, Events } from '../view';
import { Observer } from 'rxjs';

describe('DesktopSkeleton', () => {
  let observer: Mocked<Observer<Events>>;

  beforeEach(() => {
    observer = {
      complete: vi.fn(),
      error: vi.fn(),
      next: vi.fn(),
    };
  });

  describe('no dialog', () => {
    it('renders', () => {
      const app = render(
        <DesktopSkeleton
          body={<div>body</div>}
          dialog={undefined}
          footer={<div>footer</div>}
          header={<div>header</div>}
          observer={observer}
          />
      );
      expect(app.baseElement).toMatchSnapshot();
      app.unmount();  
    });
  });

  describe('with dialog', () => {
    let app: RenderResult;
  
    beforeEach(() => {
      app = render(
        <DesktopSkeleton
          body={<div>body</div>}
          dialog={<div>dialog</div>}
          footer={<div>footer</div>}
          header={<div>header</div>}
          observer={observer}
          />
      );
    });
  
    afterEach(() => {
      app.unmount();
    });
  
    it('renders', () => {
      expect(app.baseElement).toMatchSnapshot();
    });
  
    it('closes on close', () => {
      const closeButton = app.getByLabelText(CLOSE_BUTTON_ARIA_LABEL);
      expect(observer.next).not.toHaveBeenCalled();
      closeButton.click();
      expect(observer.next).toHaveBeenCalledTimes(1);
      expect(observer.next).toHaveBeenCalledWith({
        type: 'skeleton_close_dialog',
      });
    });
  });

});