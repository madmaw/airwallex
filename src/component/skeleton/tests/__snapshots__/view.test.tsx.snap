// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DesktopSkeleton > no dialog > renders 1`] = `
<body>
  <div>
    <div
      class="_container_141b3a"
    >
      <div
        class="_header_141b3a"
      >
        <div>
          header
        </div>
      </div>
      <div
        class="_body_141b3a"
      >
        <div>
          body
        </div>
      </div>
      <div
        class="_footer_141b3a"
      >
        <div>
          footer
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`DesktopSkeleton > with dialog > renders 1`] = `
<body>
  <div />
  <div>
    <div
      class="_container_141b3a"
    >
      <div
        class="_header_141b3a"
      >
        <div>
          header
        </div>
      </div>
      <div
        class="_body_141b3a"
      >
        <div>
          body
        </div>
      </div>
      <div
        class="_footer_141b3a"
      >
        <div>
          footer
        </div>
      </div>
      <div
        class="_overlay_141b3a"
      >
        <div
          class="_dialog_141b3a"
        >
          <div>
            dialog
          </div>
          <button
            aria-label="Close"
            class="_closeButton_141b3a"
          >
            x
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
